src/lib.rs

```diff
@@ -915,12 +916,15 @@ impl LocalQueue {
                             let waker = cx.waker();
                             move |slot| {
                                 let mut slot = slot.borrow_mut();
-                                let qaw = slot.as_mut().expect("missing local queue");
-
-                                // If we've been replaced, just ignore the slot.
-                                if !Arc::ptr_eq(&qaw.queue, queue) {
-                                    return;
-                                }
```

Comment:
...I just saw this now. I have to start work now; I'll fix this later, don't merge this yet.